File: server/app/schemas/mocktest.py
from pydantic import BaseModel, Field


class GenerateMockTestRequest(BaseModel):
    pdfId: str
    difficulty: str = "medium"
    numberOfQuestions: int = Field(default=10, ge=1, le=50)
    topic: str | None = None
    timeLimitMinutes: int | None = None


class MockQuestion(BaseModel):
    id: str
    question: str
    options: list[str]


class MockTestResponse(BaseModel):
    testId: str
    questions: list[MockQuestion]


class SubmitMockTestRequest(BaseModel):
    testId: str
    answers: list[dict]
    timeTakenSeconds: int = 0


class SubmitMockTestResponse(BaseModel):
    score: float
    totalQuestions: int
    correctAnswers: int
    wrongAnswers: int
    accuracy: float
    weakTopics: list[str] = []
    strongTopics: list[str] = []

File: client/src/types/mocktest.ts
export interface MockQuestion {
  id: string;
  question: string;
  options: string[];
}

export interface MockTest {
  testId: string;
  questions: MockQuestion[];
}

export interface MockTestAnswer {
  questionId: string;
  selectedOption: number;
}

export interface MockTestResult {
  score: number;
  totalQuestions: number;
  correctAnswers: number;
  wrongAnswers: number;
  accuracy: number;
  weakTopics: string[];
}